lib/client/Client.js

```diff
@@ -336,23 +336,33 @@ export default class Client {
 
   /**
    * Destroy the {@link Mailbox} related to the given _id_ on the server.<br />
-   * This will issue a {@link Client#setMailboxes} JMAP request under the hoods, passing the correct options.
+   * This will issue a {@link Client#destroyMailboxes} request under the hoods, passing _[id]_ option.
    *
    * @param id {String} The id of the mailbox to destroy.
    *
-   * @return {Promise} A {@link Promise} that eventually resolves to nothing if the {@link Mailbox} was destroyed successfully.
+   * @return {Promise} A {@link Promise} that eventually resolves to JMAP response. See {@link Client#setMailboxes} for response data structure.
    *
-   * @see http://jmap.io/spec.html#destroying-mailboxes
    */
   destroyMailbox(id) {
     Utils.assertRequiredParameterIsPresent(id, 'id');
 
-    return this.setMailboxes({destroy: [id]})
-      .then(response => {
-        if (response.destroyed.indexOf(id) < 0) {
-          throw new Error('Failed to destroy ' + id + ', the reason is: ' + (response.notDestroyed[id] || 'missing'));
-        }
-      });
```

Comment:
Please don't !
We do want to keep this logic here. Also you did factor exactly the part you should have not :)

Have a look at how destroyMessage delegates to destroyMessages please